CodingTracker.DreamFXX/DatabaseManager.cs

```diff
@@ -0,0 +1,140 @@
+                                            WHERE Date = '{date}'";
+                connection.Execute(sql);
+            }
+        }
+
+        public void UpdateRecord(string? oldDate, string? newDate, string? startTime, string? endTime, string? duration)
+        {
+            using (var connection = new SqliteConnection(_connectionString))
+            {
+                connection.Open();
+                
+                    if (startTime is null && endTime is null)
+                    {
+                        connection.Execute($@"UPDATE MyCodingTracker
+                                                SET Date = '{newDate}'
+                                                WHERE Date = '{oldDate}'");
+                    }
+                    else if (newDate is null)
+                    {
+                        connection.Execute($@"UPDATE MyCodingTracker
+                                                SET StartTime = '{startTime}', EndTime = '{endTime}', Duration = '{duration}'
+                                                WHERE Date = '{oldDate}'");
+                    }
+                    else
+                    {
+                        connection.Execute($@"UPDATE MyCodingTracker
+                                               SET Date = '{newDate}', StartTime = '{startTime}', EndTime = '{endTime}', Duration = '{duration}'
+                                               WHERE Date = '{oldDate}'");
+                    }
+            }
+        }
+
+        public List<CodingSession> ReadFromDb()
+        {
+            List<CodingSession> codingSessions = new();
+
+            using (var connection = new SqliteConnection(_connectionString))
+            {
+                connection.Open();
+
+                string sqlQuery = @"SELECT * FROM MyCodingTracker";
+                using (var command = new SqliteCommand(sqlQuery, connection))
+                using (var reader = command.ExecuteReader())
+
+                    if (reader.HasRows)
+                    {
+                        while (reader.Read())
+                        {
+                            var model = new CodingSession
+                            {
+                                Id = Convert.ToInt32(reader["Id"]),
+                                Date = reader["Date"].ToString()!,
+                                StartTime = reader["StartTime"].ToString()!,
+                                EndTime = reader["EndTime"].ToString()!,
+                                Duration = reader["Duration"].ToString()!
+                            };
+
+                            codingSessions.Add(model);
+                        }
+                    }
+                    else
+                    {
+                        AnsiConsole.MarkupLine("[red]No records found in the database.[/]");
+                    }
+            }
+            return codingSessions;
+        }
+
+        public CodingSession? ReadSingleRecord(string? date)
```

Comment:
🔴 **ADO.NET**

Can you change to use Dapper instead of ADO.NET?
https://www.learndapper.com/dapper-query